rae_hw/src/rae_motors.cpp

```diff
@@ -183,57 +189,96 @@ namespace rae_hw
     {
         while (_running)
         {
-            usleep(1);
+            
             int currA = enAPin.get_value();
             int currB = enBPin.get_value();
             State currS{currA, currB};
+            bool direction=true; //true is forwards, false is backwards
             int count = prevCount;
-            if (currS == Clockwise)
+            if (currS == Clockwise && currS != prevState)
             {
                 if (prevState == Rest)
                 {
-                    encDirection = true;
+                    count++;
+                    direction=true;
+                }
+                else if (prevState == Halfway)
+                {
+                    count--;
+                    direction=false;
+                }
+                else{
+                   // std::cout << "We are missing ticks, adding or removing 2 based on previous state  " << pwmPin << "Prev and current state" << prevState.A << prevState.B << currS.A << currS.B << std::endl;
```

Comment:
There's few comments here as well

Reply:
Didnt git pull before pushing so git just ignored my changes for some reason